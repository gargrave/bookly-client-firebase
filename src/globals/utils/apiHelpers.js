// @flow
import type { FbDocRef } from '../../modules/core/flowtypes';

async function getDocRef(
  db: any,
  table: string,
  id: string,
): FbDocRef {
  return db.collection(table).doc(id);
}

export {
  getDocRef,
};
