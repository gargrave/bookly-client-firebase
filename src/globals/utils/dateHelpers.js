export const toUnixTimestamp = (date, defaultValue) => {
  if (!date) {
    if (defaultValue === null || defaultValue === undefined) {
      return Date.now();
    }
    return defaultValue;
  }

  if (date._isAMomentObject) {
    return date.unix();
  }

  if (typeof date === 'number') {
    return date;
  }

  return defaultValue;
};

export const toUnixMs = (date) =>
  (date && date * 1000) || 0;
