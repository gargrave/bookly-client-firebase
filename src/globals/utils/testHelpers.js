import React from 'react';
import { shallow } from 'enzyme';

export class ComponentBuilder {
  constructor(component, defaultProps) {
    this.component = component;
    this.defaultProps = defaultProps;
  }

  shallowGetComponent(extraProps = {}) {
    const props = {
      ...this.defaultProps,
      ...extraProps,
    };
    return shallow(<this.component {...props} />);
  }
}
