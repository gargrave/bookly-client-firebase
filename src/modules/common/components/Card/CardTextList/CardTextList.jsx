// @flow
import React, { Fragment } from 'react'
import { array } from 'prop-types'

import styles from './CardTextList.css'

type CardTextListItem = {
  title?: string,
  value?: string,
}

type Props = {
  textList: CardTextListItem[],
}

const CardTextList = ({ textList }: Props) => (
  <Fragment>
    {textList.map((item, i) => (
      <p key={i} className={styles.cardTextItem}>
        {item.title && <strong>{item.title}: </strong>}
        {item.value}
      </p>
    ))}
  </Fragment>
)

CardTextList.propTypes = {
  textList: array.isRequired,
}

export default CardTextList
