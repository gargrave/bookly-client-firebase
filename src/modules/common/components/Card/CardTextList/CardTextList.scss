@import '../../../../../styles/_colors';
@import '../../../../../styles/_spacing';

.cardTextItem {
  color: $textColorLight;
  margin-bottom: $spacingSm; // Shoelace override
}
