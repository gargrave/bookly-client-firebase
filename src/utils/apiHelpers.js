// @flow
import type { FbDocRef } from '../constants/flowtypes';

async function getDocRef(
  db: any,
  table: string,
  id: string,
): FbDocRef {
  return db.collection(table).doc(id);
}

export {
  getDocRef,
};
