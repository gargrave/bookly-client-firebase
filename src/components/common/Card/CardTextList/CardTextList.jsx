// @flow
import React, { Fragment } from 'react';
import { array } from 'prop-types';

import type { CardTextListItem } from '../types';

import { buildClasses } from '../../../../globals/utils/cssHelpers';

type Props = {
  textList: CardTextListItem[],
};

const CardTextList = ({
  textList,
}: Props) => {
  return (
    <Fragment>
      {
        textList.map((item, i) => {
          return (
            <p key={i} className={buildClasses('card__text')}>
              {item.title && <strong>{item.title}: </strong>}
              {item.value}
            </p>
          );
        })
      }
    </Fragment>
  );
};

CardTextList.propTypes = {
  textList: array.isRequired,
};

export default CardTextList;
