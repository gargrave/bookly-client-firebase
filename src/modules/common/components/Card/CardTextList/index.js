import CardTextList from './CardTextList'

export default CardTextList
