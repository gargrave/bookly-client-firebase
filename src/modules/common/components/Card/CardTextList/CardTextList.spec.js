import { ComponentBuilder } from '../../../../../utils/testHelpers'

import CardTextList from './CardTextList'

const textList = [
  { title: '1', value: 'one' },
  { title: '2', value: 'two' },
  { title: '3', value: 'three' },
]
const defaultProps = {
  textList,
}

const builder = new ComponentBuilder(CardTextList, defaultProps)

describe('CardTextList', () => {
  let component

  describe('basic rendering', () => {
    it('renders correctly', () => {
      component = builder.shallowGetComponent()
      expect(component).toMatchSnapshot()
      expect(component.children().length).toBe(textList.length)
    })
  })
})
