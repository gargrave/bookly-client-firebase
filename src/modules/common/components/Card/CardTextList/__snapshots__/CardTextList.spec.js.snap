// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTextList basic rendering renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardTextList
    textList={
      Array [
        Object {
          "title": "1",
          "value": "one",
        },
        Object {
          "title": "2",
          "value": "two",
        },
        Object {
          "title": "3",
          "value": "three",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Styled(div)>
          <strong>
            1
            : 
          </strong>
          one
        </Styled(div)>,
        <Styled(div)>
          <strong>
            2
            : 
          </strong>
          two
        </Styled(div)>,
        <Styled(div)>
          <strong>
            3
            : 
          </strong>
          three
        </Styled(div)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <strong>
              1
              : 
            </strong>,
            "one",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "1",
                ": ",
              ],
            },
            "ref": null,
            "rendered": Array [
              "1",
              ": ",
            ],
            "type": "strong",
          },
          "one",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <strong>
              2
              : 
            </strong>,
            "two",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2",
                ": ",
              ],
            },
            "ref": null,
            "rendered": Array [
              "2",
              ": ",
            ],
            "type": "strong",
          },
          "two",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <strong>
              3
              : 
            </strong>,
            "three",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "3",
                ": ",
              ],
            },
            "ref": null,
            "rendered": Array [
              "3",
              ": ",
            ],
            "type": "strong",
          },
          "three",
        ],
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Styled(div)>
            <strong>
              1
              : 
            </strong>
            one
          </Styled(div)>,
          <Styled(div)>
            <strong>
              2
              : 
            </strong>
            two
          </Styled(div)>,
          <Styled(div)>
            <strong>
              3
              : 
            </strong>
            three
          </Styled(div)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <strong>
                1
                : 
              </strong>,
              "one",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "1",
                  ": ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "1",
                ": ",
              ],
              "type": "strong",
            },
            "one",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <strong>
                2
                : 
              </strong>,
              "two",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2",
                  ": ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "2",
                ": ",
              ],
              "type": "strong",
            },
            "two",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <strong>
                3
                : 
              </strong>,
              "three",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "3",
                  ": ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "3",
                ": ",
              ],
              "type": "strong",
            },
            "three",
          ],
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
